src/views/tickets/Screenshot.js

```diff
@@ -1,18 +1,63 @@
 import React, { Component } from 'react';
-import { Image, Segment, Grid } from 'semantic-ui-react';
+import {
+  Image,
+  Segment,
+  Divider,
+  Icon,
+  Button,
+  Modal,
+} from 'semantic-ui-react';
 import { getTicket, getArtifactURL } from '../../utils/api.js';
 
+import 'pure-react-carousel/dist/react-carousel.es.css';
+import { CarouselProvider, Slide, Slider } from 'pure-react-carousel';
+import CustomDotGroup from '../../components/CustomDotGroup.js';
+
 export default class Screenshot extends Component {
   state = {
-    height: 0,
-    width: 0,
+    carouselLength: 1,
+    singleScreenshot: true,
   };
+  carousel = [];
+
   async getTicket() {
+    // always get full page screenshot and add it to the carousel
+    let image = (
+      <Slide index={0} width="100%">
+        <Image
+          alt="fullpage screenshot"
+          src={getArtifactURL(this.props.ticketID, 'screenshotFull.png')}
+          bordered
+          centered
+        />
+      </Slide>
+    );
+    this.carousel.push(image);
+
     try {
-      let value = await getTicket(this.props.ticketID);
+      let ticket = await getTicket(this.props.ticketID);
+      const length =
+        this.state.carouselLength + ticket.ticket.screenshots.length;
+      if (length > 1) {
+        this.setState({ singleScreenshot: false });
```

Comment:
Could we use `this.state.carouselLength` in place of this extra variable? I might be missing something here.

Reply:
hahaha yes let me fix that, I'm not used there being a state.

Reply:
Thanks! I was thinking more about `singleScreenshot` though - do we need that where it's used, or can we check `carouselLength > 1`?

Reply:
I'll fix that too!